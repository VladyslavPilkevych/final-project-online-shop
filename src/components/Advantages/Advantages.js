/* eslint-disable max-len */
import React from 'react';
import styles from './Advantages.module.scss';
import accountIcon from '../../assets/icons/account-icon.png';
import supportIcon from '../../assets/icons/support-icon.png';
import savingIcon from '../../assets/icons/saving-icon.png';

function Advantages() {
  return (
    <div className={styles.advantagesWrapper}>
      <div className={styles.advantagesCard}>
        <img src={accountIcon} className={styles.advantagesIcon} alt="advantages-icon" />
        <h3 className={styles.advantagesTitle}>Product Support</h3>
        <p className={styles.advantagesText}>Up to 3 years on-site warranty available for your peace of mind.</p>
      </div>
      <div className={styles.advantagesCard}>
        <img src={supportIcon} className={styles.advantagesIcon} alt="advantages-icon" />
        <h3 className={styles.advantagesTitle}>Personal Account</h3>
        <p className={styles.advantagesText}>With big discounts, free delivery and a dedicated support specialist.</p>
      </div>
      <div className={styles.advantagesCard}>
        <img src={savingIcon} className={styles.advantagesIcon} alt="advantages-icon" />
        <h3 className={styles.advantagesTitle}>Amazing Savings</h3>
        <p className={styles.advantagesText}>Up to 70% off new Products, you can be sure of the best price.</p>
      </div>
    </div>
  );
}

export default Advantages;
