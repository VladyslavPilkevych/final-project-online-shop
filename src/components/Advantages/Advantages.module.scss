@import '../../assets/styles/global.module.scss';



.advantagesWrapper {
    max-width: 1055px;
    margin: 0 auto;
    display: flex;
    justify-content: space-between;
    padding-top: 50px;
    column-gap: 5px;
}

.advantagesCard {
    display: grid;
    grid-template-columns: 1fr;
    row-gap: 18px;
    justify-items: center;
    text-align: center;
    max-width: 260px;
}

.advantagesTitle {
    font-family: $fonts;
    font-weight: 700;
    font-size: 18px;
    line-height: 132.5%;
}

.advantagesText {
    font-family: $fonts;
    font-weight: 400;
    font-size: 14px;
    line-height: 140%;
    opacity: 0.7;
}

@media screen and (min-device-width: 566px) and (max-device-width: 768px) {
    .advantagesIcon {
        height: 46px;
        width: 45px;
    }

    .advantagesTitle {
        font-size: 14px;
    }

    .advantagesText {
        font-size: 12px;
    }
}

@media screen and (max-device-width: 565px) {
    .advantagesWrapper {
        display: grid;
        grid-template-columns: 1fr;
        justify-items: center;
        row-gap: 20px;
    }

    .advantagesIcon {
        height: 46px;
        width: 45px;
    }
}