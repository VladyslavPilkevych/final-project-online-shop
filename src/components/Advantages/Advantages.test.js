/* eslint-disable react/react-in-jsx-scope */
import { render } from '@testing-library/react';
import Advantages from './Advantages';

describe('Snapshot advantages should render', () => {
  test('Snapshot advantages is rendered', () => {
    const { asFragment } = render(<Advantages />);
    expect(asFragment()).toMatchSnapshot();
  });
});