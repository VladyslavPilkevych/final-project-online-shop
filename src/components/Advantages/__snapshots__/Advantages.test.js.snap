// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot advantages should render Snapshot advantages is rendered 1`] = `
<DocumentFragment>
  <div
    class="advantagesWrapper"
  >
    <div
      class="advantagesCard"
    >
      <img
        alt="advantages-icon"
        class="advantagesIcon"
        src="account-icon.png"
      />
      <h3
        class="advantagesTitle"
      >
        Product Support
      </h3>
      <p
        class="advantagesText"
      >
        Up to 3 years on-site warranty available for your peace of mind.
      </p>
    </div>
    <div
      class="advantagesCard"
    >
      <img
        alt="advantages-icon"
        class="advantagesIcon"
        src="support-icon.png"
      />
      <h3
        class="advantagesTitle"
      >
        Personal Account
      </h3>
      <p
        class="advantagesText"
      >
        With big discounts, free delivery and a dedicated support specialist.
      </p>
    </div>
    <div
      class="advantagesCard"
    >
      <img
        alt="advantages-icon"
        class="advantagesIcon"
        src="saving-icon.png"
      />
      <h3
        class="advantagesTitle"
      >
        Amazing Savings
      </h3>
      <p
        class="advantagesText"
      >
        Up to 70% off new Products, you can be sure of the best price.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
